### ISSUE TITLE
"forwardPorts" conflicts with docker-compose port mapping

### ISSUE THREAD

**Issue by Foreverflying** [microsoft/vscode-dev-containers#344]
- VSCode Version: 1.45.1
- Local OS Version: MacOS 10.15.4
- Name of Dev Container Definition with Issue: forwardPorts conflicts with docker-compose port mapping

If we open a browser by auto launching after build (or by clicking an HTTP link maybe), which accesses port 8080, the vscode will try to map 8080 to a random localhost port and open the link with that port.

If we want it to mapping to a fixed port, we need to add a line `forwardPorts: [8080]` in devcontainer.json. Then when container get created, vscode will add a "8080:8080" mapping to the container, and open the link to port 8080 with port 8080.

**however, when devcontainer.json indicates to start a container with a docker-compose.yml file, and the port mapping 8080:8080 is already set, the forwardPorts config seems not work anymore, I guess it tries to do the mapping but failed due to the port is in use.**

In this case, when you open a link to port 8080, the vscode will map it to a random port rather than 8080, even the forwardPorts is set to [8080].

Steps to Reproduce:

1. add forwardPorts in file .devcontainer/devcontainer.json, mapping the same ports in docker-compose.yml as well.
2. launch a browser by a url with port 8080.

**Comment by Chuxel** [comment]
Yes, if the port is already in use locally - which would be the case here - forwarding works but uses an alternate port.  For example, you may be running a server on your local machine for that is taking up the port, so it uses another one.  That's the expected behavior.  Is this causing a problem for you?
